prow/cmd/deck/main.go

```diff
@@ -1222,16 +1244,51 @@ func handleProwJob(prowJobClient prowv1.ProwJobInterface) http.HandlerFunc {
 }
 
 // canTriggerJob determines whether the given user can trigger any job.
-func canTriggerJob(user string, cfg *config.RerunAuthConfig) bool {
+func canTriggerJob(user string, pj prowapi.ProwJob, cfg *config.RerunAuthConfig, cli prowgithub.RerunClient) (bool, error) {
 	if cfg.AllowAnyone {
-		return true
+		return true, nil
 	}
 	for _, u := range cfg.AuthorizedUsers {
 		if prowgithub.NormLogin(u) == prowgithub.NormLogin(user) {
-			return true
+			return true, nil
+		}
+	}
+	jobPermissions := pj.Spec.RerunPermissions
```

Comment:
Instead of duplicating the logic for the global (deck config) and job specific rerun permissions we should unify the types and refactor these checks into a method.